gopherage/pkg/cov/junit/calculation/coveragelist.go

```diff
@@ -17,46 +17,35 @@ limitations under the License.
 package calculation
 
 import (
-	"fmt"
 	"path"
 	"strings"
 )
 
 // CoverageList is a collection and summary over multiple file Coverage objects
 type CoverageList struct {
 	*coverage
-	Group            []coverage
-	summaryCompleted bool
+	Group []coverage
 }
 
 // CovList constructs new (file) Group Coverage
 func newCoverageList(name string) *CoverageList {
 	return &CoverageList{
-		coverage:         &coverage{Name: name},
-		Group:            []coverage{},
-		summaryCompleted: false,
+		coverage: &coverage{Name: name},
+		Group:    []coverage{},
 	}
 }
 
-// Percentage returns the percentage of statements covered
-func (covList *CoverageList) Percentage() string {
-	return fmt.Sprintf("%.1f", covList.Ratio()*100)
-}
-
 // Ratio summarizes the list of coverages and returns the summarized ratio
 func (covList *CoverageList) Ratio() float32 {
-	covList.Summarize()
+	covList.summarize()
 	return covList.coverage.Ratio()
 }
 
-// Summarize summarizes all items in the Group and stores the result
-func (covList *CoverageList) Summarize() {
-	if !covList.summaryCompleted {
-		for _, item := range covList.Group {
-			covList.nCoveredStmts += item.nCoveredStmts
-			covList.nAllStmts += item.nAllStmts
-		}
-		covList.summaryCompleted = true
+// summarize summarizes all items in the Group and stores the result
+func (covList *CoverageList) summarize() {
```

Comment:
This removed the reuse check, but the results are still not actually _correct_ if this gets called more than once. While in `Ratio` this doesn't matter if the function is called with no changes to `Group`, if there are changes the results will be surprising, and if some future feature uses the absolute values of `nCoveredStmts` or `nAllStmts`

Please explicitly zero out everything before this calculation.

```suggestion
func (covList *CoverageList) summarize() {
	covList.nCoveredStmts = 0
	covList.nAllStmts = 0
```

Reply:
thanks, nice catch